### ISSUE TITLE
Modify Airsonci to read artists from tags instead of filesystem

### ISSUE THREAD

**Issue by joshuaboniface** [airsonic/airsonic#499]
This is a minor annoyance I've always had with Sub-/Libre-/Airsonic. It appears to read and populate the Arists list (the left bar) based on the folder names on the filesystem. This is OK for 95% of artist names, but for some like `AC/DC` which contain invalid characters, the filesystem folder name (in this case for me, `AC_DC`) might not be the same as the artist name and this can get annoying in the UI.

I recommend adding an option (if it already exists, I don't know where) or changing the default behaviour to read the artist names from the tags of each file and group them appropriately, rather than assuming the directory name is the artist name.

Possibly ties in with #164 since implementing that should enable this sort of behaviour by default, but I think it would be really useful regardless.

It's worth noting that the database itself (in the table `artist`) shows the correct name.

**Comment by jooola** [comment]
I would like to add a little something :
This option could be great but I think that we need to add way more option about the filesystem structure !

For example end the left side bar I don't have artist but genre so when I browse a genre I get the last FM bio from the reggae which isn't an artiste !
So I think that we could create a separate issue with a listing of tools/option that we can add to airsonic that will enhance the filesystem engine !

I totally agree with Joshua but we may want to she something bigger !

**Comment by joshuaboniface** [comment]
@jooola Oh that's a really good idea - allow multiple "sorting" on the left bar.

Looking through the database, all the "right" information seems to be there in the various tables (e.g. `artist`, `genre`, etc.) so it should be really simple to have the sidebar query the database for a list rather than assume "they're artists and what's on the FS is right".

**Comment by joshuaboniface** [comment]
So I did a bit more digging and experimenting with this.

From the looks of `airsonic-main/src/main/java/org/airsonic/player/dao/ArtistDao.java`, this information is pulled out of the DB table "artists", which makes no sense - if it was, I wouldn't be seeing (again, example) `AC_DC` when the `artist` table record is `AC/DC`. The one place I did notice the wrong information was in the `media_file` table, in the `DIRECTORY` entry for the artist parent folder - it had the incorrect name. However even after adjusting that, there's no difference in the UI through multiple restarts (of the VM and of tomcat8 alone). Just for fun, I also tried "renaming" another artist (`Yes`) to `YesPlease` and restarting - same thing, the entry in the left-hand artist list doesn't change. So either this is being cached somehow and not reflecting the database changes, or it's looking somewhere completely different.

Anything beyond that is a little out of my scope as a non-developer, but hopefully this points towards the answer.

**Comment by gerroon** [comment]
I personally like the file system approach because I structure it around folders, so if you are guys are going to implement this please do so in a way that is possible to switch between these methods.

**Comment by lbesnard** [comment]
The left panel has been a bit of an annoyance for me too. I have set up my folder structure as genre/albumArtist/album , but some of my friends/users would prefer browsing the music based on the id3 tags of their choice, which makes total sense imo. So this needs to be flexible so each user can choose his left panel

**Comment by joshuaboniface** [comment]
@gerroon I do too, which is actually how I found the bug in the first place (`AC_DC` since `AC/DC` isn't a valid POSIX directory name by itself).

However it seems like all this data is stored in the DB anyways, and from my cursory check of the code above it should be reading from the DB too, it just doesn't seem to. I've tried a bit more to dig around for the cause but have been unsuccessful, but in general this seems like a positive thing to work towards for other reasons as @lbesnard and @jooola have mentioned!

**Comment by pacija** [comment]
Let me chime in on this topic as well.

I think it would be nice to have a possibility of having file and folder names not only POSIX compatible, but also compatible with most file systems, including ftp-friendly filenames on a FAT32. For example `[:SITD:]` is POSIX compatible folder name, but I can't rsync it to FAT32. This could be done by sourcing everything from ID3 tags, and not folders.

Next issue - compilations. I guess most people prefer to have their compilations similar to this:
```
Various Artists\
  Album\
   01 - Artist A - Track X
   02 - Artist B - Track Y
   03 - Artist C - Track Z
```
In above folder structure, artists which appear on compilations only never return search results in "Artists" category, which is not only inconvenient for a user, but also unfair to the artist - some of my favourite tracks are produced by artists who have only one track released, on a compilation. These, most frequently fresh, up and coming, but sometimes obscure, old, forgotten, artists need more, not less promotion than those who have dozens of albums in their discographies. Not having artist page also prevents fetching info from last.fm. Now, I haven't tried, but I guess I could work around by modifying folder structure like this:
```
Artist A\
  Album\
    01 - Track X
Artist B\
  Album\
    02 - Track Y
Artist C\
  Album\
    03 - Track Z

```
This seems inappropriate to me, however - to have tracks from single release scattered over multiple folders just for the purpose of satisfactory display in a web app.

I feel a bit bad about only pointing out issues in airsonic, never sending any patches. Lately I am intensively testing quite a number of selfhosted media servers, and i like airsonic best by far. I don't know anything about writing java web apps, but I think I'd be able to catch up fast if I had someone to show me around. I mean, I wrote short text adventure in BASIC on my C64 back in a day :)

**Comment by khogeland** [comment]
I've always hated the left bar for this reason. My music directory is huge and poorly organized, but it's all tagged correctly. I've never cared to organize it better because every other music player UI I've used (ncmpcpp, Foobar, Plex) organizes by tags, and I never have to think about the file system.

**Comment by Lightsockie** [comment]
Oh, I like the bar as. My music directory is massive and immaculately organized, both in tagging and structure. But the lack of agreement between "Artist" and "Album Artist" reigning king means that there's a lot of conflicts between which would be the better choice in a tag-based sorter. Plus some VA albums are best laid under, say, a master OST directory/listing, instead of the specific (likely) one-off artist who sang that track

So basically I'm seconding @gerroon. Please have the option to keep it as-is if you do add a new sorting method

**Comment by albertoMartinsen** [comment]
Hi,
As a new user here running away from iTunes and its Apple Music service, and testing this as a docker, I must agree with this thread. Sorry, I'm not a programmer, so I can't contribute with code, but I like good apps with good usability and clever solutions and Airsonic is fantastic, exactly what I was looking for. The only issue I have is this one, as a AC/DC fan it's really sad to see its name destroyed :(. I think most people have good ID3 tagged libraries and if not, there're solutions for it too (to auto-tag it) so it shouldn't be a big deal.

**Comment by RandomNinjaAtk** [comment]
+1 For this also! I try to make my files linux friendly, as in no spaces at all in file names. And this completely breaks the airsonic web interface. It won't look up properly who the artist is because it attempts to use the folder name instead of the tags in the files. When viewing the actual album, the track tags appear properly. But album names, and artist names for browsing are based on folder names and looks completely broken.

DSub app has an option to browse using tags instead of the folder/file naming. And it works perfectly in the app.

**Comment by rickylford** [comment]
I'd like to add a +1 for how annoying the sidebar is. AC/DC should not show up as AC+DC because that isn't the artist's name. Browsing by folder is a horrible way to go when you can just poll the database to get the ID3 tag.

**Comment by bubblesnout** [comment]
+1 for this, while I can see why having your library listed based on the filesystem I feel like more users would find it useful to have the tagged artist names instead. An option to choose would be nice.

**Comment by Faeb35** [comment]
+1 would be highly appreciated

**Comment by lbesnard** [comment]
i have to say I'm not a big fan of this auto-stale bot feature

**Comment by cfw34683** [comment]
Why is MP3 tagging such a big deal? Airsonic obviously dont need it.. this issue has staled out 6 times.. the creator dont wanna do it.. and have never chimed in.// this gives me no hope of fixing an error i have

**Comment by albertoMartinsen** [comment]
> the creator dont wanna do it.. and have never chimed in.// this gives me no hope of fixing an error i have

That's why my solution is called Plex. It solved my issue, I have an amazing library on my server with very little effort there and an amazing player both on desktop and mobile

**Comment by pacija** [comment]
> @cfw34683 Why is MP3 tagging such a big deal? Airsonic obviously dont need it.. this issue has staled out 6 times.. the creator dont wanna do it.. and have never chimed in.// this gives me no hope of fixing an error i have

Unfortunately, this particular repository had last commit more than 6 months ago, which probably means this, or any other issues reported here, won't be resolved in this particular piece of software.

> @albertoMartinsen That's why my solution is called Plex. It solved my issue, I have an amazing library on my server with very little effort there and an amazing player both on desktop and mobile

I hope you are not shilling for free and that you at least got free license for promoting them.

As for people who look for free and open alternatives, which are hopefully actively working on fixing issues reported here, here are some interesting *sonic projects:

https://github.com/deluan/navidrome/ (has web interface for listening)
https://github.com/sentriz/gonic (web interface for administration, needs client for listening)

As for me, I still like airsonic best even with all the bugs and issues :)

**Comment by albertoMartinsen** [comment]
> I hope you are not shilling for free and that you at least got free license for promoting them.

So, the only reason for sharing an alternative is if they pay me? I'm just a happy customer.
